File: backend/app/api/v1/alert_history.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.repositories.alert_history_repository import AlertHistoryRepository
from app.schemas.alert_history import AlertHistoryEntry

router = APIRouter(prefix="/alert-history", tags=["alert-history"])


@router.get("/", response_model=list[AlertHistoryEntry])
async def list_alert_history(
    limit: int = Query(100, ge=1, le=500),
    offset: int = Query(0, ge=0),
    severity: str | None = Query(None),
    state: str | None = Query(None),
    db: AsyncSession = Depends(get_db),
) -> list[AlertHistoryEntry]:
    return await AlertHistoryRepository(db).get_recent(
        limit=limit, offset=offset, severity=severity, state=state
    )

File: frontend/src/api/alertHistory.ts
import type { AlertHistoryEntry } from "../types/alert";
import { apiFetch } from "./client";

export const getAlertHistory = (
  limit = 100,
  offset = 0,
  severity?: string,
  state?: string,
) => {
  const params = new URLSearchParams({ limit: String(limit), offset: String(offset) });
  if (severity) params.set("severity", severity);
  if (state) params.set("state", state);
  return apiFetch<AlertHistoryEntry[]>(`/alert-history/?${params}`);
};
